### PR TITLE
feat(dataTable): saving method optimization for colDefs

### PR DESCRIPTION
**Pull request by egrycs**

### CODE REVIEW

**Review by sanyisuveges — APPROVED** [review]

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@i-cell/data-table",
-  "version": "3.0.4",
+  "version": "3.0.6",
```
Comment (welsen) [inline package.json:3]: this should be the same as the current version published (3.0.5)

File: src/data-table.component.ts [hunk 2]
```diff
@@ -483,17 +465,32 @@ export class DataTableComponent implements OnInit, OnDestroy, OnChanges {
...
+    });
+  }
+
+  public handleColumnSelectionChange($event) {
+    this.columnSelection.clear();
+    const selectedColumnsWithAlwaysVisibleColumns = [
+      ...new Set([
+        ...$event.value,
+        ...(Array.isArray(this.columnSettings)
+          ? (this.columnSettings as DataTableColumnDefinition[]).filter((i) => !i.hideable && i.visible)
```
[the remark this answers is not in the document]
Comment (welsen) [reply src/data-table.component.ts:489]: 👍

**Review by welsen — CHANGES_REQUESTED** [review]
Please add the new property `actionColumn` to the column settings section of the `README.adoc`

**Review by welsen — APPROVED** [review]
🦅

**Pull request merged**